### PR TITLE
Add i18n support (wip)

### PR DESCRIPTION
**Pull request by m1ga**
Hi :wave: 
found your project in the Hacktoberfest section! Started to add i18n support (https://github.com/eventzimmer/site/issues/19) using vue-i18n.
It will add a select box with de/en and will change the strings on the fly.

It is my first time using Vue so I'm still learning to get my way around the project :smile: But it is working already but can still be optimized!

PS:
Hoffentlich nicht allzu schlimm, ich passe noch einiges an :wink:

### CODE REVIEW

**Comment by Fohlen** [comment]
Hey @m1ga thanks for tackling this PR! ✨ 

Apparently I hadn't watched this repo (which I thought was the default setting for repo owners) 🦆 
I will check this PR and let you know what I think and/or how it could be improved!

**Comment by m1ga** [comment]
@Fohlen Nice! I still want to change the imprint/privacy part (loading custom files in custom languages).

**Comment by Fohlen** [comment]
My first impression of this PR is quite good, here are my two cents:

- you can use the `store` to save the language setting. The state of the store is automatically persisted to `localStorage` and restored upon loading, thus no need for `vue-cookies`. Just use `store.state.lang` once loaded.
- we could use [vue-router](https://router.vuejs.org/) to assemble the different parts (impress / privacy pages). I am unsure if it is worth the effort. What do you think? 
- the language button should be where the other buttons are. I can handle the CSS as you like

I do really like your translations, head's up 👍 
Greetings from rainy Tuebingen! 🌧

File: src/store.js [hunk 1]
```diff
@@ -119,6 +123,9 @@ export default new Vuex.Store({
     },
     categories () {
       return new Set(Object.values(CATEGORIES_PER_LOCATION).reduce((acc, val) => acc.concat(val), [])) // NOTE: This can be replaced with Array.flat
+    },
```
Comment (Fohlen) [inline src/store.js]: Unless we're planning to return a different result for `lang` we don't really need a getter, we can use `store.state.lang`.

**Review by Fohlen — CHANGES_REQUESTED** [review]
Looks almost good to go. What's your decision about the `imprint` / `privacy` pages? Should we just create different links in the footer or should we use `vue-router` ?

**Comment by m1ga** [comment]
> Looks almost good to go. What's your decision about the `imprint` / `privacy` pages? Should we just create different links in the footer or should we use `vue-router` ?

I'll have to have a closer look what vue-route does and how it works. Different links would be quicker and if it is only de/en it should be fine.

BTW:
I've moved the Select-Box to the right and applied the normal style. I'm not sure if they should look like this at all:
Screenshot_20191003_170256
or if my Firefox/Fedora 30 is breaking the style :smile: So I think you can style it better later on (e.g. make it smaller?)

**Comment by Fohlen** [comment]
> > Looks almost good to go. What's your decision about the `imprint` / `privacy` pages? Should we just create different links in the footer or should we use `vue-router` ?
> 
> I'll have to have a closer look what vue-route does and how it works. Different links would be quicker and if it is only de/en it should be fine.
> 
> BTW:
> I've moved the Select-Box to the right and applied the normal style. I'm not sure if they should look like this at all:
> Screenshot_20191003_170256
> or if my Firefox/Fedora 30 is breaking the style 😄 So I think you can style it better later on (e.g. make it smaller?)

I believe this is expected behaviour, but I can make some amendments to make it more compelling.
Let's do it like this:

- I'll sweep over your PR and add some finishing touches here and there
- You'll add links to the english imprint / privacy (just copy the german one)
- I'll translate the rest of the imprint and privacy, as I have time
- Once done, I merge the PR

**Comment by m1ga** [comment]
sounds like a good plan! I'll have a look at it tomorrow.

**Comment by Fohlen** [comment]
Closing in favour of #31

**Pull request closed without merging**